### PR TITLE
Node visual refresh v3

### PR DESCRIPTION
**Pull request by OliverEGreen**
### Purpose

L6eEl6DACg

This PR combines several small changes made to the node visual refresh scope and addresses the following JIRA tickets: 

**DYN-4035 CodeBlock font size smaller than existing**
- Code block font has been matched to the Figma mockup. 
- Output port has been increased in width from 16px to 20px.

**DYN-4036 Gap between top and bottom of code in codeblock different?**
- Code block has been visually balanced to have equal top/bottom margins for >1 lines. Avalon editor treats single-line code blocks slightly differently, meaning we cannot have completely balanced single-line code blocks. 

**DYN-4100 Right-click on node does not trigger context menu**
- Original functionality has been restored. Right-click now opens the context menu at all zoom levels. A further minor bug has been resolved here, ensuring the node is selected and set as the DataContext before context the menu is displayed.

**DYN-4088 No gap between new Node Redesign and preview bubble**
- A 6 pixel margin has been added and there is always a visible gap between the node and its PreviewBubble now. 

### Summary of Changes:
- Code block output port height goes from 15.72px to 16.435 pixels.
- Code block font is now Source Code Pro (Regular) at 13px.
- +/- buttons on variable input nodes have been restyled as grey with a blue rollover effect.
- Node options button has been restyled as grey with a blue rollover effect. 
- Code block margins have been visually balanced. 
- Node port styling logic has changed to new design.
- Node port 'value marker' is now always visible, and restyles to match whether ports are resolved/unresolved.
- List@Level menu has been restyled to new design.
- Color palette node was restyled to be more space-efficient.
- Node header was restyled to be more space-efficient in its width.
- Node frozen style has rounded corners to match node shape.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.

### Reviewers

@QilongTang @mjkkirschner 

### FYIs

@Amoursol

### CODE REVIEW

File: src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml:2071]: Given this is null, do we have to set this?

File: src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml:2102]: same comment here

File: src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml
Comment (OliverEGreen) [reply src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml]: Seems to work fine without; have removed.

File: src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml
Comment (OliverEGreen) [reply src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml]: Seems to work fine without; have removed.

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml:117]: Can you explain about this change here?

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml:176]: same here

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml
Comment (OliverEGreen) [reply src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml:117]: Yeah I changed this from being the config value to something hard-coded. The visual mockup shows the port as slightly smaller than the full port height in order to introduce a bit of a visual gap between ports.

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml:367]: Should we also define this color somewhere?

**Review by QilongTang — COMMENTED** [review]
Some comments then LGTM

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml
Comment (OliverEGreen) [reply src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml:367]: Done - Changed to the very similar PrimaryCharcoal300Brush

**Review by QilongTang — COMMENTED** [review]
LGTM with one last comment about usage of `CodeBlockOutputPortHeightInPixels` because I see the slightly larger height in pixels still referenced in a couple places in code, I wonder if this definition is still needed?

**Comment by OliverEGreen** [comment]
> LGTM with one last comment about usage of `CodeBlockOutputPortHeightInPixels` because I see the slightly larger height in pixels still referenced in a couple places in code, I wonder if this definition is still needed?

It's still used, so the Config value of 16.345px is set to match the line-to-line height in the code block editor.
Visually we're displaying the ports as a little less than this, 14px, to create the gap, but they're both still being used.

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml
Comment (OliverEGreen) [reply src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml:117]: We still use this in a few places. There's an invisible grid that's the full line-to-line height which needs to be this exact value in the config, but the light grey rectangle we display is slightly less than this for graphics' sake.

**Comment by QilongTang** [comment]
@OliverEGreen There is one regression on this PR `DynamoCoreWpfTests.ListAtLevelInteractionTests.KeepListStructureUpdatesPortViewCorrectly`, please address and after that I will approve

File: src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs
Comment (saintentropy) [inline src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs:764]: Can these colors be stored as a resource or at least defined statically in this view model.  This was @mjkkirschner observation that this might be a memory leak issue as we keep re-allocating the colors

Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs:764]: This is an issue we encountered before - when running a many NUNit tests, none of the SharedDictionaryManager dictionaries (containing these colours) would load in. Attempting to reference them would cause an exception. They've been hard-coded for now to unblock progress but it's not a clean solution.

Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs:764]: you can recycle the colors objects - avoid allocating them over and over again.

**Review by QilongTang — APPROVED** [review]
Will wait to merge until all checks pass

**Pull request merged**
